lib/ansible/modules/network/checkpoint/cp_address_range_facts.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/python
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: cp_address_range_facts
+short_description: Get address-range objects facts on Checkpoint over Web Services API
+description:
+  - Get address-range objects facts on Checkpoint devices.
```

Comment:
@chkp-orso IMHO all the get calls should be moved under `lib/ansible/plugins/lookup/*`, also in all the get modules the code seem to be duplicated with little change, please move the functionality to  lookup plugins and maintain a common piece where user can just give input of relevant `get` params and based on respective input the get call is fired and give the response as implemented currently here as a module.

Reply:
1. what do u mean by common code? there is only parameters and documentation in the modules.
2. In addition the facts mosule Riki wrote are also *not* in `lib/ansible/plugins/lookup/*`
3. can u please give an example of one module under ```7lib/ansible/plugins/lookup/*```

Reply:
@chkp-orso by common code I meant that most of the `*_facts` module can be moved under `plugins/lookup` as `checkpoint_facts.py`, please refer the code diff for ref:
<img width="1441" alt="Screenshot 2019-05-31 at 10 43 05 AM" src="https://user-images.githubusercontent.com/26872141/58683156-004c2e80-8391-11e9-8235-9f5bc4e54253.png">
So, moving the `facts` logic to one single file will help you and the user both as the user can give input `fact` type and get the facts for the respective inputted param, and you'll have to maintain only single file for any `*_facts` related CP query.

Reply:
Also, you can use [NIOS lookup plugin](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/lookup/nios.py#L27)  as a reference to write the `checkpoint_facts` plugin.

Reply:
@justjais ohh thanks it looks like it will help us also.
but what to do about the different parameters for different facts module (e.g. host_facts and network_facts have different parameters)